Modules/Registration/Common/include/itkImageToImageMetric.h

```diff
@@ -600,15 +604,15 @@ class ITK_TEMPLATE_EXPORT ImageToImageMetric : public SingleValuedCostFunction
   /**
    * \class MultiThreaderWorkUnitInfoImageToImageMetricWrapper
    * This helper local class is used to extract information from the
-   * MultiThreaderType::WorkUnitInfo info type
+   * MultiThreaderBase::WorkUnitInfo info type
    * Do not allow inheritance for objects that are intended for static_cast<void *>
    * \ingroup ITKRegistrationCommon
    */
   class MultiThreaderWorkUnitInfoImageToImageMetricWrapper final
   {
   public:
     MultiThreaderWorkUnitInfoImageToImageMetricWrapper(const void * workunitInfoAsVoid)
-      : m_WorkUnitInfo(static_cast<const typename MultiThreaderType::WorkUnitInfo *>(workunitInfoAsVoid))
+      : m_WorkUnitInfo(static_cast<const MultiThreaderBase::WorkUnitInfo *>(workunitInfoAsVoid))
```

Comment:
Note that the `typename` keyword is not necessary for `MultiThreaderBase::WorkUnitInfo`   😃